### PR TITLE
Fix SpanBench

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Make benchmark class public, add namespace. Tweak iteration counts
to very roughly 1 second per xunit-perf iteration.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@joperezr @ahsonkhan PTAL
cc @dotnet/jit-contrib 

Resolves the ongoing mystery of why this benchmark hasn't been producing perf results: when I originally wrote it I did not make the benchmark class public.

Local testing using run-xunit-perf shows reasonable results now.

File: tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs [hunk 1]
```diff
@@ -14,13 +14,20 @@
 [assembly: OptimizeForBenchmarks]
 [assembly: MeasureInstructionsRetired]
 
-class Tests
+namespace Span
+{
+
+public class Tests
```
Comment (ahsonkhan) [inline tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs]: Can you rename this from Tests to something like SpanBench (or more descriptive and unique). Makes it easier to search, especially in benchview.

Comment (AndyAyersMS) [inline tests/src/JIT/Performance/CodeQuality/Span/SpanBench.cs]: Sure

**Comment by ahsonkhan** [comment]
Looks good to me :)

**Comment by AndyAyersMS** [comment]
Meant to ping @jorive....

**Comment by jorive** [comment]
@AndyAyersMS Changes look good. Optionally, you could change the benchmarks inner iterations to look more like this:
```
[Benchmark(InnerIterationCount = 10000)]
[MemberData(nameof(InputData))]
public static void TestBenchmark(string[] args)
{
    foreach (BenchmarkIteration iter in Benchmark.Iterations)
    {
        using (iter.StartMeasurement())
        {
            for (int i = 0; i < Benchmark.InnerIterationCount; ++i)
```
As-is, the results will display on BenchView like this: `Span \ SpanBench \ TestFillAllSpan`
I would suggest something more like: `Span \ Benchmarks \ TestFillAll` (remove the repetition)
What do you think?

**Comment by AndyAyersMS** [comment]
The final `Span` suffix is there to distinguish the span cases from the corresponding array cases -- that is there is both a `TestFillAllSpan` and a `TestFillAllArray`.

This test can be run as a perf test two ways -- under xunit perf and also as a standalone exe. I generally haven't generally used inner iteration count for the tests I write to maintain symmetry with the exe-driven portion of the testing.

Happy to change the class name to `Benchmarks` if you think it is an improvement.

**Comment by ahsonkhan** [comment]
@AndyAyersMS, I believe SpanBench is fine.

There is already a method: 
```C#
    static double Bench(Action f)
```

error CS0542: 'Bench': member names cannot be the same as their enclosing type [D:\j\workspace\debug_windows---17180f1b\tests\src\JIT\Performance\CodeQuality\Span\SpanBench.csproj]

**Comment by ahsonkhan** [comment]
@AndyAyersMS, you probably don't want to name your class "Benchmark"

SpanBench.cs(203,45): error CS0117: 'Benchmark' does not contain a definition for 'Iterations' [D:\j\workspace\checked_windo---f6dc6fe4\tests\src\JIT\Performance\CodeQuality\Span\SpanBench.csproj]

It gets confused with Benchmark attribute for the performance tests.
```C#
foreach (var iteration in Benchmark.Iterations)
```

**Comment by AndyAyersMS** [comment]
Odd that I don't see any of these errors building locally... anyways, back to SpanBench it is.

**Comment by AndyAyersMS** [comment]
Arm emulator build failure, seems widespread since roughly midnight.
```
07:23:12 Copied 205 artifacts from "dotnet_coreclr » master » release_windows_nt_bld" build number 1215
07:23:12 ERROR: Unable to find project for artifact copy: dotnet_corefx/master/linuxarmemulator_softfp_cross_release
07:23:12 This may be due to incorrect project name or permission settings; see help for project name in job configuration
```
looks like this artifact comes from a corefx build, last good run has:
```
Copied 1 artifact from "dotnet_corefx » master » linuxarmemulator_softfp_cross_debug" build number 116
```
Root cause is likely this change which broke arm builds in CoreFX:
https://github.com/dotnet/corefx/pull/16378

**Comment by AndyAyersMS** [comment]
@ahsonkhan I'll abandon this since your #9790 has incorporated these changes already.

**Comment by hqueue** [comment]
@dotnet-bot test Linux ARM Emulator Cross Release Build
@dotnet-bot test Linux ARM Emulator Cross Debug Build

**Comment by ahsonkhan** [comment]
> I'll abandon this since your #9790 has incorporated these changes already.

Sounds good. Close this PR?

**Pull request closed without merging**